app/data_model/answer_store.py

```diff
@@ -203,35 +204,18 @@ def filter(self, group_id=None, block_id=None, answer_id=None, group_instance=No
                 if v is not None:
                     matches = matches and answer[k] == v
             if matches:
-                filtered.append(answer)
+                if escape_values:
+                    answer = answer.copy()
+                    if isinstance(answer['value'], str):
+                        answer['value'] = escape(answer['value'])
+                filtered.append(answer['value'] if flat_values else answer)
```

Comment:
I'm not sure `flat_values` makes sense in this method - if the filter is returning values for more than one answer id, won't it just be an array of values with no way of identifying which answer they are for?

Reply:
@ajmaddaford true. It was more of a feature creep for convenience. It probably makes more sense to keep it specific to the helpers in Question.py while it's not required elsewhere.